tracy-client/src/plot.rs

```diff
@@ -6,6 +6,89 @@ use crate::Client;
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct PlotName(pub(crate) &'static str);
 
+/// The format of a plot to be shown in the Tracy profiler UI.
+#[derive(Debug, Hash, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Default)]
+#[non_exhaustive]
+pub enum PlotFormat {
+    /// Values will be displayed as plain numbers. This is the default.
+    #[default]
+    Number,
+
+    /// Values will be displayed as byte counts, showing as kilobytes, megabytes, etc.
+    Memory,
+
+    /// Values will be shown as a percentage (with 100 being equal to 100%).
+    Percentage,
+
+    /// Values will be shown as watts.
+    Watts,
+}
+
+/// The style of lines of a plot, shown in the Tracy profiler UI.
+#[derive(Debug, Hash, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Default)]
+#[non_exhaustive]
+pub enum PlotLineStyle {
+    /// Lines will be stepped (ie look like a staircase).
+    Stepped,
+
+    /// Lines will be smooth (interpolating a line between two values).
+    #[default]
+    Smooth,
+}
+
+/// Configuration for how a plot appears in the Tracy profiling UI.
```

Comment:
Worth adding a code example here. I found it to be pretty common for people to struggle to figure out how to construct stuff, when the construction is not a `fn new()` (even if it is as simple as using `default()` instead.) A plain example would help with that.